tests/binning_test.py

```diff
@@ -1044,3 +1044,12 @@ def test_group_automatic_groups_works_with_string_coord():
     assert sc.identical(
         table.group('key').coords['key'], sc.array(dims=['key'], values=['a', 'b', 'c'])
     )
+
+
+def test_explicit_groups_with_mismatching_dtype():
+    label = sc.arange('row', 10, dtype='int32')
+    table = sc.DataArray(sc.ones(dims=['row'], shape=[10]), coords={'label': label})
+    groups = sc.arange('label', 8, dtype='int64')
+    assert sc.identical(table.group(groups).coords['label'], groups)
+    groups[-1] += 1
```

Comment:
Not sure I understand what this is doing, but otherwise looks fine.

Reply:
Good point, added a comment. Clearer?